File: php-aihint/examples/scoring_example.php
<?php
require_once __DIR__ . '/../src/scoring/index.php';

use AIHint\Scoring\TrustScoringEngine;
use AIHint\Scoring\ScoringResult;

echo "AiHint Trust Scoring System - PHP Example\n";
echo "========================================\n\n";

// Example 1: Basic scoring
echo "Example 1: Basic Website Scoring\n";
echo "--------------------------------\n";

try {
    $engine = new TrustScoringEngine();
    $result = $engine->scoreWebsite('https://github.com');
    
    echo "URL: " . $result->url . "\n";
    echo "Trust Score: " . number_format($result->finalScore, 3) . "\n";
    echo "Trust Level: " . $result->trustLevel->name . "\n";
    echo "Confidence: " . number_format($result->confidence, 3) . "\n";
    echo "Security Score: " . number_format($result->securityScore, 3) . "\n";
    echo "Reputation Score: " . number_format($result->reputationScore, 3) . "\n";
    echo "Compliance Score: " . number_format($result->complianceScore, 3) . "\n";
    
    if (!empty($result->warnings)) {
        echo "\nWarnings:\n";
        foreach ($result->warnings as $warning) {
            echo "  - $warning\n";
        }
    }
    
    if (!empty($result->errors)) {
        echo "\nErrors:\n";
        foreach ($result->errors as $error) {
            echo "  - $error\n";
        }
    }
    
} catch (Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
}

echo "\n" . str_repeat('=', 50) . "\n\n";

// Example 2: Custom configuration
echo "Example 2: Custom Configuration\n";
echo "-------------------------------\n";

$customConfig = [
    'ssl' => [
        'timeout' => 15
    ],
    'headers' => [
        'timeout' => 15
    ],
    'malware' => [
        'timeout' => 15,
        'google_safe_browsing' => [
            'enabled' => false, // Set to true and add API key for real checks
            'api_key' => 'YOUR_API_KEY_HERE'
        ],
        'virustotal' => [
            'enabled' => false, // Set to true and add API key for real checks
            'api_key' => 'YOUR_API_KEY_HERE'
        ],
        'phishtank' => [
            'enabled' => true // Free service
        ]
    ],
    'weights' => [
        'security' => 0.5,      // 50% weight for security
        'reputation' => 0.3,    // 30% weight for reputation
        'compliance' => 0.2     // 20% weight for compliance
    ]
];

try {
    $engine = new TrustScoringEngine($customConfig);
    $result = $engine->scoreWebsite('https://stackoverflow.com');
    
    echo "URL: " . $result->url . "\n";
    echo "Trust Score: " . number_format($result->finalScore, 3) . "\n";
    echo "Trust Level: " . $result->trustLevel->name . "\n";
    echo "Method: " . $result->method . "\n";
    
} catch (Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
}

echo "\n" . str_repeat('=', 50) . "\n\n";

// Example 3: Batch scoring
echo "Example 3: Batch Scoring\n";
echo "------------------------\n";

$urls = [
    'https://github.com',
    'https://stackoverflow.com',
    'https://example.com'
];

$results = [];

foreach ($urls as $url) {
    try {
        echo "Scoring: $url\n";
        $result = $engine->scoreWebsite($url);
        $results[] = [
            'url' => $result->url,
            'score' => $result->finalScore,
            'level' => $result->trustLevel->name,
            'confidence' => $result->confidence
        ];
    } catch (Exception $e) {
        echo "  Error: " . $e->getMessage() . "\n";
        $results[] = [
            'url' => $url,
            'score' => 0.0,
            'level' => 'ERROR',
            'confidence' => 0.0
        ];
    }
}

echo "\nBatch Results:\n";
foreach ($results as $result) {
    echo sprintf("%-30s | Score: %5.3f | Level: %-10s | Confidence: %5.3f\n",
        $result['url'],
        $result['score'],
        $result['level'],
        $result['confidence']
    );
}

echo "\n" . str_repeat('=', 50) . "\n\n";

// Example 4: JSON output
echo "Example 4: JSON Output\n";
echo "----------------------\n";

try {
    $result = $engine->scoreWebsite('https://github.com');
    echo $result->toJson() . "\n";
} catch (Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
}

echo "\nDone!\n";

File: examples/scoring_example.py
#!/usr/bin/env python3
"""
AiHint Trust Scoring Example

Demonstrates the complete trust scoring system with all three phases:
- Phase 1: Core Security Metrics (SSL, headers, malware)
- Phase 2: Reputation Signals (domain age, incidents, reputation)
- Phase 3: Content & Compliance (privacy, contact, legal)
"""

import asyncio
import json
import sys
import os
from datetime import datetime, timezone

# Add the parent directory to the path so we can import aihint
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))

from aihint.scoring import TrustScoringEngine


async def main():
    """Main example function."""
    print("AiHint Trust Scoring System Example")
    print("=" * 50)
    
    # Configuration for the scoring engine
    config = {
        'ssl': {
            'timeout': 10
        },
        'headers': {
            'timeout': 10
        },
        'malware': {
            'timeout': 10,
            'google_safe_browsing': {
                'enabled': False,  # Set to True and add API key for real checks
                'api_key': None
            },
            'virustotal': {
                'enabled': False,  # Set to True and add API key for real checks
                'api_key': None
            },
            'phishtank': {
                'enabled': False  # Set to True for real checks
            }
        },
        'reputation': {
            'timeout': 10
        },
        'domain_age': {
            'timeout': 10
        },
        'incidents': {
            'timeout': 10
        },
        'privacy': {
            'timeout': 10
        },
        'contact': {
            'timeout': 10
        },
        'compliance': {
            'timeout': 10
        }
    }
    
    # Initialize the scoring engine
    engine = TrustScoringEngine(config)
    
    # Example URLs to score
    test_urls = [
        "https://example.com",
        "https://github.com",
        "https://stackoverflow.com",
        "https://httpbin.org",
        "https://www.google.com"
    ]
    
    print(f"Scoring {len(test_urls)} websites...")
    print()
    
    # Score each URL
    results = []
    for url in test_urls:
        print(f"Scoring: {url}")
        try:
            result = await engine.score_website(url)
            results.append(result)
            
            # Display results
            print(f"  Final Score: {result.final_score:.3f}")
            print(f"  Trust Level: {result.trust_level.name}")
            print(f"  Confidence: {result.confidence:.3f}")
            print(f"  Security: {result.security_score:.3f}")
            print(f"  Reputation: {result.reputation_score:.3f}")
            print(f"  Compliance: {result.compliance_score:.3f}")
            
            if result.warnings:
                print(f"  Warnings: {', '.join(result.warnings)}")
            
            if result.errors:
                print(f"  Errors: {', '.join(result.errors)}")
            
            print()
            
        except Exception as e:
            print(f"  Error: {e}")
            print()
    
    # Generate summary report
    print("Summary Report")
    print("=" * 50)
    
    if results:
        avg_score = sum(r.final_score for r in results) / len(results)
        avg_confidence = sum(r.confidence for r in results) / len(results)
        
        print(f"Average Score: {avg_score:.3f}")
        print(f"Average Confidence: {avg_confidence:.3f}")
        print()
        
        # Categorize by trust level
        trust_levels = {}
        for result in results:
            level = result.trust_level.name
            if level not in trust_levels:
                trust_levels[level] = []
            trust_levels[level].append(result.url)
        
        for level, urls in trust_levels.items():
            print(f"{level}: {len(urls)} websites")
            for url in urls:
                print(f"  - {url}")
            print()
    
    # Save detailed results to file
    output_file = "scoring_results.json"
    with open(output_file, 'w') as f:
        # Convert results to serializable format
        serializable_results = []
        for result in results:
            serializable_results.append({
                'url': result.url,
                'final_score': result.final_score,
                'trust_level': result.trust_level.name,
                'confidence': result.confidence,
                'security_score': result.security_score,
                'reputation_score': result.reputation_score,
                'compliance_score': result.compliance_score,
                'warnings': result.warnings,
                'errors': result.errors,
                'timestamp': result.timestamp.isoformat(),
                'method': result.method,
                'detailed_metrics': result.detailed_metrics
            })
        
        json.dump(serializable_results, f, indent=2)
    
    print(f"Detailed results saved to: {output_file}")
    
    # Demonstrate individual scorer usage
    print("\nIndividual Scorer Examples")
    print("=" * 50)
    
    from aihint.scoring.scorers import SSLTLSValidator, SecurityHeadersAnalyzer
    
    # SSL/TLS example
    print("SSL/TLS Validation Example:")
    ssl_validator = SSLTLSValidator()
    ssl_score, ssl_metrics = await ssl_validator.score("https://example.com")
    print(f"  Score: {ssl_score:.3f}")
    print(f"  Details: {json.dumps(ssl_metrics, indent=2)}")
    print()
    
    # Security Headers example
    print("Security Headers Analysis Example:")
    headers_analyzer = SecurityHeadersAnalyzer()
    headers_score, headers_metrics = await headers_analyzer.score("https://example.com")
    print(f"  Score: {headers_score:.3f}")
    print(f"  Headers Analyzed: {headers_metrics.get('headers_analyzed', 0)}")
    print(f"  Headers Present: {headers_metrics.get('headers_present', 0)}")
    print()


if __name__ == "__main__":
    asyncio.run(main())
